engine/evd.ml

```diff
@@ -1474,8 +1480,11 @@ let restrict evk filter ?candidates ?src evd =
   | None -> candidate_evars
   | Some _ -> Evar.Set.add evk' candidate_evars
   in
-  let evd = { evd with undf_evars = EvMap.add evk' evar_info' undf_evars;
-    defn_evars; last_mods; evar_names; evar_flags; candidate_evars }
+  let evd =
+    { evd with
+      evar_counter = evd.evar_counter + 1;
```

Comment:
Here you do. But where do you do it in the `new_pure_evar` function?

Reply:
add_with_name increments the counter